packages/side-nav/src/styles/vaadin-side-nav-base-styles.js

```diff
@@ -8,39 +8,35 @@ import { css } from 'lit';
 import { sharedStyles } from './vaadin-side-nav-shared-base-styles.js';
 
 const sideNav = css`
-  @layer base {
-    :host {
-      white-space: nowrap;
-      touch-action: manipulation;
-    }
+  :host {
+    white-space: nowrap;
+    touch-action: manipulation;
+  }
 
-    [part='label'] {
-      align-self: start;
-      display: flex;
-      align-items: center;
-      justify-content: start;
-      gap: var(--vaadin-side-nav-item-gap, var(--vaadin-gap-container-inline));
-      padding: var(--vaadin-side-nav-item-padding, var(--vaadin-padding-container));
-      font-size: var(--vaadin-side-nav-label-font-size, 0.875em);
-      font-weight: var(--vaadin-side-nav-label-font-weight, 500);
-      color: var(--vaadin-side-nav-label-color, var(--vaadin-color-subtle));
-      line-height: var(--vaadin-side-nav-label-line-height, inherit);
-      border-radius: var(--vaadin-side-nav-item-border-radius, var(--vaadin-radius-m));
-      touch-action: manipulation;
-    }
+  [part='label'] {
+    align-self: start;
+    display: flex;
+    align-items: center;
+    justify-content: start;
+    gap: var(--vaadin-side-nav-item-gap, var(--vaadin-gap-container-inline));
+    padding: var(--vaadin-side-nav-item-padding, var(--vaadin-padding-container));
+    font-size: var(--vaadin-side-nav-label-font-size, 0.875em);
+    font-weight: var(--vaadin-side-nav-label-font-weight, 500);
+    color: var(--vaadin-side-nav-label-color, var(--vaadin-color-subtle));
+    line-height: var(--vaadin-side-nav-label-line-height, inherit);
+    border-radius: var(--vaadin-side-nav-item-border-radius, var(--vaadin-radius-m));
+    touch-action: manipulation;
   }
 `;
 
 export const sideNavStyles = [sharedStyles, sideNav];
 
 export const sideNavSlotStyles = css`
-  @layer vaadin.base {
-    vaadin-side-nav:has(vaadin-icon[slot='prefix']) {
-      --_has-prefix-icon: '';
-    }
+  :where(vaadin-side-nav:has(vaadin-icon[slot='prefix'])) {
+    --_has-prefix-icon: '';
+  }
 
-    vaadin-side-nav-item:has(> vaadin-icon[slot='prefix'])::part(link) {
-      --_has-prefix-icon:;
-    }
+  :where(vaadin-side-nav-item:has(> vaadin-icon[slot='prefix']))::part(link) {
```

Comment:
Not sure if we need `:where` here but I added it to align with other components using `SlotStylesMixin`.